### PR TITLE
GHA Migration

### PR DESCRIPTION
**Pull request by LucianoCaravajal**
# Split Synchronizer

## Extra Notes

### CODE REVIEW

**Comment by Israphel** [comment]
Remember the "unstable" workflow. That pushes a container to the docker-dev repository, for every branch (non PRs) which aren't master. And the tag is a short hash.

```yaml
  - name: unstable
    if: NOT (type = pull_request) AND NOT (branch = master)

```

```yaml
    - stage: unstable
      language: minimal
      services:
        - docker
      before_script:
        - COMMIT=$(git rev-parse --short HEAD)
      script:
        - docker build -t ${ARTIFACTORY_REGISTRY}/split-synchronizer:${COMMIT} .
      deploy:
        provider: script
        script: >-
          echo "${ARTIFACTORY_PASSWORD}" | docker login -u "${ARTIFACTORY_USER}" --password-stdin "${ARTIFACTORY_REGISTRY}" &&
          docker push ${ARTIFACTORY_REGISTRY}/split-synchronizer
        skip_cleanup: true
        on:
          all_branches: true
```

File: .github/workflows/unstable.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Get short hash
+        run: echo "SHORT_SHA=git rev-parse --short HEAD" >> $GITHUB_ENV
+      - name: Docker Build and Push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: splitio-docker-dev.jfrog.io/${{ github.event.repository.name }}:$SHORT_SHA
```
Comment (Israphel) [inline .github/workflows/unstable.yml:29]: that's not how you reference a gihub actions env var

**Review by Israphel — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            -Dsonar.host.url=${{ secrets.SONARQUBE_HOST }}
+            -Dsonar.projectName=${{ github.event.repository.name }}
+            -Dsonar.projectKey=${{ github.event.repository.name }}
+            -Dsonar.projectVersion=${{ env.VERSION }}
+            -Dsonar.exclusions='**/*_test.go,**/vendor/**,**/testdata/*'
+            -Dsonar.go.coverage.reportPaths=coverage.out
+            -Dsonar.links.ci="https://github.com/splitio/${{ github.event.repository.name }}/actions"
+            -Dsonar.links.scm="https://github.com/splitio/${{ github.event.repository.name }}"
+
+  deploy-to-stage:
```
Comment (Israphel) [inline .github/workflows/cd.yml:78]: split-synchronizer is not something we deploy.

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    name: Deploy to Stage
+    runs-on: ubuntu-latest
+    needs: push-docker-image
+    steps:
+      - name: Deploy to R2D2 in stage
+        run: |
+          curl --header "Content-Type: application/json" \
+            --request PUT \
+            --data '{	"service": "${{ github.event.repository.name }}", "environment": "stage", "tag": "${{ github.sha }}", "token": "${{ secrets.SLACK_TOKEN }}"}' \
+            https://r2d2.split-stage.io/deployment
```
Comment (Israphel) [inline .github/workflows/cd.yml:88]: end of line missing

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            -Dsonar.projectName=${{ github.event.repository.name }}
+            -Dsonar.projectKey=${{ github.event.repository.name }}
+            -Dsonar.projectVersion=${{ env.VERSION }}
+            -Dsonar.exclusions='**/*_test.go,**/vendor/**,**/testdata/*'
+            -Dsonar.go.coverage.reportPaths=coverage.out
+            -Dsonar.links.ci="https://github.com/splitio/${{ github.event.repository.name }}/actions"
+            -Dsonar.links.scm="https://github.com/splitio/${{ github.event.repository.name }}"
+            -Dsonar.pullrequest.key=${{ github.event.pull_request.number }}
+            -Dsonar.pullrequest.branch=${{ github.event.pull_request.head.ref }}
+            -Dsonar.pullrequest.base=${{ github.event.pull_request.base.ref }}
```
Comment (Israphel) [inline .github/workflows/ci.yml:54]: end of line missing

File: .github/workflows/unstable.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Get short hash
+        run: echo "SHORT_SHA=git rev-parse --short HEAD" >> $GITHUB_ENV
+      - name: Docker Build and Push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: splitio-docker-dev.jfrog.io/${{ github.event.repository.name }}:${{ env.SHORT_SHA }}
```
Comment (Israphel) [inline .github/workflows/unstable.yml:29]: end of line missing

File: .github/workflows/cd.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
+name: cd
+on:
+  push:
+    branches:
+      - main
+      - master
```
Comment (Israphel) [inline .github/workflows/cd.yml:6]: leave a line before jobs

**Review by Israphel — CHANGES_REQUESTED** [review]

File: .github/workflows/unstable.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Login to Artifactory
+        uses: docker/login-action@v1
+        with:
+          registry: splitio-docker-dev.jfrog.io
+          username: ${{ secrets.ARTIFACTORY_DOCKER_USER }}
+          password: ${{ secrets.ARTIFACTORY_DOCKER_PASS }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Get short hash
+        run: echo "SHORT_SHA=git rev-parse --short HEAD" >> $GITHUB_ENV
```
Comment (Israphel) [inline .github/workflows/unstable.yml:23]: `run: echo "SHORT_SHA=$(git rev-parse --short HEAD)" >> $GITHUB_ENV`

**Review by Israphel — DISMISSED** [review]

**Review by mmelograno — APPROVED** [review]

**Review by nikoe14 — APPROVED** [review]

**Review by Israphel — APPROVED** [review]

**Pull request merged**